CEN-59

manuel-salvador opened the pull request:
## Resumen
Se creo un TempLayout (Layout temporal) y se uso eso para desarrollar el Dashboard, dentro de este se usan diferentes componentes creados como los charts y el mock de lastOrders para poder utilizar estos datos en la tabla del final de la pagina.
Además en el componente AreaChart.vue se usa una funcion para crear los  datos aleatoriamente y que se tengan datos ficticios de ventas de 1 año entero

## Cambios Realizados

- Modificado el router para crear ruta /dashboard
- Modificado DashboardView
- Modificado App.vue para que no aparezca el header default junto con RouterView
- Modificado App.vue para crear variables de colores que se usan repetidamente
- Creado componente TempLayout
- Instalado paquete apexcharts
- Instalado paquete vue3-apexcharts
- Modificado main.ts para que se pueda usar <apexchart> en cualquier lugar
- Creado componente AreaChart
- Creado componente DonutChart
- Creado carpeta /utils
- Creado funcion generateData en /utils
- Creado carpeta /mocks
- Creado mock lastOrdersMock en /mocks
- Modificado script dev para que abra una nueva pestaña al iniciar el servidor de desarrollo

# Comprobación de Calidad

- [ ] Se han añadido pruebas unitarias relevantes.
- [ ] Se han ejecutado todas las pruebas existentes.
- [ ] Otros: _(especificar)_

## Revisores Asignados

@brayanlaiton0116 
@saracast909

manuel-salvador committed:
Initial commit

manuel-salvador committed:
recommended extensions

manuel-salvador committed:
docs: update readme

manuel-salvador committed:
feat: added vuetify

manuel-salvador committed:
docs: Update README.md

manuel-salvador committed:
Merge pull request #1 from manuel-salvador/manuel-salvador-patch-1

docs: Update README.md

manuel-salvador committed:
fix(project): :art: added and configured @mdi/font

manuel-salvador committed:
refactor(project): :wrench: deleted unnecesary comment

manuel-salvador committed:
Merge pull request #2 from manuel-salvador/material-icons

fix(project): :art: added and configured @mdi/font

manuel-salvador committed:
feat(project): :white_check_mark: added pre-commit checks eslint rules in staged files before commit

manuel-salvador committed:
Merge pull request #3 from manuel-salvador/precommit

feat(project): :white_check_mark: added pre-commit checks eslint rule…

manuel-salvador committed:
docs(project): :memo: added 2 PR templates, bug and feature

manuel-salvador committed:
Merge pull request #4 from manuel-salvador/pr-templates

docs(project): :memo: added 2 PR templates, bug and feature

manuel-salvador committed:
fix(project): :art: deleted main.css and removed unnecessary styles from base.css

manuel-salvador committed:
fix(project): :art: deleted main.css and removed unnecessary styles from base.css

manuel-salvador committed:
Merge pull request #5 from manuel-salvador/fix-base-css

fix(project): :art: deleted main.css and removed unnecessary styles from base.css

manuel-salvador committed:
feat(dashboard): created Dashboard page with TempLayout, added 2 charts packages

manuel-salvador committed:
feat(dashboard): created Dashboard page with TempLayout, added 2 charts packages

manuel-salvador committed:
feat(dashboard): created Dashboard page with TempLayout, added 2 charts packages

saracast909 commented on src/components/AreaChart.vue:
```diff
@@ -0,0 +1,179 @@
+  }
+
+  const today = new Date().getTime()
+  let startOfYear // Movemos la declaración fuera del bloque 'case'
```
Escribir comentario en ingles

manuel-salvador replied:
Solucionado

saracast909 commented on src/components/AreaChart.vue:
```diff
@@ -0,0 +1,179 @@
+      chart.value.zoomX(today - 365 * 24 * 60 * 60 * 1000, today)
+      break
+    case 'ytd':
+      startOfYear = new Date(today).getFullYear() // Asignamos el valor aquí
```
Comentario en ingles

manuel-salvador replied:
Solucionado

manuel-salvador committed:
refactor(Charts): :recycle: removed comments and renamed updateData function

Removed unnecessary comments because that was just for understanding in development and renamed updateData function to updateChartTimeline because it better describes what it does

brayanlaiton0116 commented on src/components/AreaChart.vue:
```diff
@@ -0,0 +1,179 @@
+  }
+
+  const today = new Date().getTime()
+  let startOfYear // Movemos la declaración fuera del bloque 'case'
```
Comentarios en ingles

manuel-salvador replied:
Esto ya fue corregido en este [commit](https://github.com/manuel-salvador/cenius/pull/7/commits/7b92b9ada9bb45279647be874b47f616ba9afab4)

brayanlaiton0116 commented on src/components/AreaChart.vue:
```diff
@@ -0,0 +1,179 @@
+    case 'one_year':
+      chart.value.zoomX(today - 365 * 24 * 60 * 60 * 1000, today)
+      break
+    case 'ytd':
```
No usar abreviaturas o contracciones en el código de case 'ytd':

manuel-salvador replied:
Corregido

brayanlaiton0116 commented on src/components/AreaChart.vue:
```diff
@@ -0,0 +1,179 @@
+        1Y
+      </button>
+
+      <button id="ytd" @click="updateData('ytd')" :class="{ active: selection === 'ytd' }">
```
no escribir abreviaturas o contracciones.

manuel-salvador replied:
Corregido

brayanlaiton0116 reviewed (changes requested):
Revisa las abreviaturas o contracciones.

brayanlaiton0116 reviewed (dismissed):
se prueba correctamente

manuel-salvador committed:
refactor(Charts): :recycle: modified abbrevations to complete words, added tooltips in area chart timeline buttons

brayanlaiton0116 reviewed (approved):
Cambios aprobados
